### ISSUE TITLE
OcMotion dataset

### ISSUE THREAD

**Issue by muses0229** [tiangexiang/OccNeRF#6]
Hi,I wonder will the OcMotion dataset be public?

**Comment by tiangexiang** [comment]
yes! Sorry for the delay, it is expected to be released before the end of this month.

**Comment by tiangexiang** [comment]
The preprocessed two OcMotion sequences along with the config files have been released. Please see README for more details! Please let me know if you encounter any problems.

**Comment by muses0229** [comment]
Thanks for sharing the dataset.Now I can't figure out how to select train dataset and eval dataset for OcMotion.The 0011_02 video has 501 images totally,and since like you choose 501 images for training.But unlike zju_mocap,there is no script to preprocess OcMotion for evaluation,how should I do?

**Comment by yejr0229** [comment]
Also there is no eval.sh script for OcMotion to eval.

**Comment by muses0229** [comment]
It seems like OcMotion only have the occluded images,we can't compare the predicted image with the unoccluded image ,so I wonder how to calculate the metrics for the predicted images?

**Comment by tiangexiang** [comment]
Yes, in real-world data, like OcMotion, there is no GT for the occluded regions. In the paper, we only computed SSIM/PSNR on the visible human parts only (masked by the binary masks). The provided OcMotion data are already preprocessed. Please let me know if you encounter any problems during evaluation.

**Comment by muses0229** [comment]
Thanks a lot!Now I try to project world 3d pts onto 2d image plane,and the results doesn't look right,seems like the translation is a bit wrong.I am not familiar with the OcMotion dataset,could you please give me some advices?

**Comment by muses0229** [comment]
![22](https://github.com/tiangexiang/OccNeRF/assets/132911839/d3d62c9f-ac36-45c7-8ac0-92949503c08a)

**Comment by muses0229** [comment]
Here is my code to project points,the R,T,K are all cameras' parameters.
    RT = torch.cat([torch.tensor(data.R.transpose()), torch.tensor(data.T).reshape(3,1)], -1)[None, None].cuda()
    xyz = torch.repeat_interleave(world_3d_pts.squeeze(0)[None, None], repeats=RT.shape[1], dim=1) 
    xyz = torch.matmul(RT[:, :, None, :, :3].float(), xyz[..., None].float()) + RT[:, :, None, :, 3:].float()
    xyz = torch.matmul(torch.tensor(data.K)[None, None][:, :, None].float().cuda(), xyz)[..., 0]
    xy = xyz[..., :2] / (xyz[..., 2:] + 1e-5)
    src_uv = xy.view(-1, *xy.shape[2:]).squeeze(0)    # (6890,2)

**Comment by tiangexiang** [comment]
Your projection script looks good, did you use the provided official SMPL translation when obtaining the `world_3d_pts`?

**Comment by muses0229** [comment]
I use the following code to transform points from smpl space to world space,where R=smpl_params['R'] ,Th=smpl_params['Th'] 
        R_inv = torch.inverse(R)
        world_src_pts = torch.matmul(smpl_src_pts, R_inv) + Th
And the extrinsics of camera have transform to world coord using these codes:
        E = apply_global_tfm_to_camera(
                E=E, 
                Rh=dst_skel_info['Rh'],
                Th=dst_skel_info['Th']) 
        R = E[:3, :3]
        T = E[:3, 3]

**Comment by yejr0229** [comment]
> Here is my code to project points,the R,T,K are all cameras' parameters. RT = torch.cat([torch.tensor(data.R.transpose()), torch.tensor(data.T).reshape(3,1)], -1)[None, None].cuda() xyz = torch.repeat_interleave(world_3d_pts.squeeze(0)[None, None], repeats=RT.shape[1], dim=1) xyz = torch.matmul(RT[:, :, None, :, :3].float(), xyz[..., None].float()) + RT[:, :, None, :, 3:].float() xyz = torch.matmul(torch.tensor(data.K)[None, None][:, :, None].float().cuda(), xyz)[..., 0] xy = xyz[..., :2] / (xyz[..., 2:] + 1e-5) src_uv = xy.view(-1, *xy.shape[2:]).squeeze(0) # (6890,2)

This script works fine for zju_mocap_refine...

**Comment by muses0229** [comment]
Sorry,I make a mistake.This is without camera global transform:
![camera_no_global](https://github.com/tiangexiang/OccNeRF/assets/132911839/c8619970-209e-4a0b-9e46-e2100e2e12b3)
This one with camera global transform:
![camera_global](https://github.com/tiangexiang/OccNeRF/assets/132911839/03058093-6061-4a15-942c-f05e1fe189e2)

**Comment by muses0229** [comment]
> Your projection script looks good, did you use the provided official SMPL translation when obtaining the `world_3d_pts`?

And if I add smpl_param['trans']  on world_3d_pts,the src_uv will be out of boundary.

**Comment by muses0229** [comment]
If I remove the global transform of cameras and project without smpl_param['Th'] like this:
      src_uv = project_3d_to_2d(data, torch.matmul(smpl_src_pts, R_inv))
I will get:
![camera_smpl_pts_with_R_inv](https://github.com/tiangexiang/OccNeRF/assets/132911839/c9f5611d-ce03-40e3-a5c6-02790582a6c8)
Seems like still missing a translation,but the provided official SMPL translation didn't work.

**Comment by tiangexiang** [comment]
Someone has found that using the visualization script from [the OcMotion repo](https://github.com/boycehbz/CHOMP?tab=readme-ov-file#visualize-3d-meshes) may give the correct rendering. Please give it a try. I will look into this issue a bit later myself :)

**Comment by muses0229** [comment]
sorry for bothering again,but my projection result still seems a bit wrong.I'm pretty sure my projection code is right because I get the same result using ocmotion's code.
Here is my code to transform points from canonical space to observation space,I remove 'Th' and get the above result but still not right.
![DZNY81JTI`2}7R}0QQ@@B_U](https://github.com/boycehbz/CHOMP/assets/132911839/88e2aeb3-61a1-4556-8280-a7e2c21df937)

**Comment by hi-zhangjie** [comment]
From my previous tests, the SMPL parameters and camera parameters are correct in both the ocmotion dataset and the preprocessed data in this repository.
![000000](https://github.com/tiangexiang/OccNeRF/assets/45507407/6aa2c9a0-5e16-43d2-8177-0977deb01be7)

**Comment by muses0229** [comment]
Thanks for sharing,I check my code and found I add wrong transpose on camera.R,now my projection result is like that:
![F2N%KB SG13W$JUTBBEK~5L](https://github.com/tiangexiang/OccNeRF/assets/132911839/33a19623-4a5e-4b96-98ec-0b1ca1049de5)
Is this looks right now?

**Comment by hi-zhangjie** [comment]
What is the id of this pic, I will check it later

**Comment by imabackstabber** [comment]
I'm a little confused by the OcMotion training and testing settings. I found that only frames from a single camera are provided. So it implies that the same picture for training is also used for testing...? Or I must be missing the frames used for testing? Where should I get them?

**Comment by muses0229** [comment]
Because OcMoton dataset didn't capture unoccluded image,I think testing is to directly rendering the input frames.

**Comment by muses0229** [comment]
> What is the id of this pic, I will check it later

I have fix this,thanks so much!

**Comment by imabackstabber** [comment]
> Because OcMoton dataset didn't capture unoccluded image,I think testing is to directly rendering the input frames.

Ok, thanks a lot! Also, I happened to encounter the same problem, which is a misalignment between SMPL mesh and humans in the 2D images using the parameters provided.(like you described in this issue), would you mind sharing your code snippet? That will help me a lot!

**Comment by ZeKai-Jiang** [comment]
> Thanks for sharing,I check my code and found I add wrong transpose on camera.R,now my projection result is like that: ![F2N%KB SG13W$JUTBBEK~5L](https://private-user-images.githubusercontent.com/132911839/310326324-33a19623-4a5e-4b96-98ec-0b1ca1049de5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAxNjI1OTAsIm5iZiI6MTcyMDE2MjI5MCwicGF0aCI6Ii8xMzI5MTE4MzkvMzEwMzI2MzI0LTMzYTE5NjIzLTRhNWUtNGI5Ni05OGVjLTBiMWNhMTA0OWRlNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcwNVQwNjUxMzBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00NjNhY2VhMjc5NWQzZTY4YjJkNTkwZjcwMTdkNTllYWQ1YmFmNjc4ZWIxNjA5ODU4NTVmNjI0YmFlZTZmZTM3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Pvcero_GwDE2yrZjQMqozVYcqEMkBSzpDOfOsOIUxvQ) Is this looks right now?

I encounter the same problem, could you explain it more clearly? Thank you very much!
